zerver/views/messages.py

```diff
@@ -822,6 +827,7 @@ def get_messages_backend(request: HttpRequest, user_profile: UserProfile,
         client_gravatar = False
 
     include_history = ok_to_include_history(narrow, user_profile)
+    narrow = filter_empty_search_terms(narrow)
```

Comment:
Would it be better to do this logic inside `narrow_parameter`, the common parsing code, instead?

Reply:
Yes, indeed!